### PR TITLE
Implement Functionality for S3BucketServerSideEncryption

### PR DESCRIPTION
**Pull request by james03160927**
## Description

Fixes https://github.com/gruntwork-io/terratest/issues/1080

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [ ] Update the docs.
- [ ] Run the relevant tests successfully, including pre-commit checks.
- [ ] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.
- [ ] Include release notes. If this PR is backward incompatible, include a migration guide.

## Release Notes (draft)

Added / Removed / Updated [X].

### CODE REVIEW

File: modules/aws/s3.go [hunk 1]
```diff
@@ -478,6 +478,37 @@ func AssertS3BucketPolicyExistsE(t testing.TestingT, region string, bucketName s
 	return nil
 }
 
+// AssertS3BucketServerSideEncryption checks if the given S3 bucket has a server side encryption configured using the given algorithm and fail the test if it does not
+func AssertS3BucketServerSideEncryption(t testing.TestingT, region string, bucketName string, algorithm types.ServerSideEncryption) {
+	err := AssertS3BucketServerSideEncryptionE(t, region, bucketName, algorithm)
+	require.NoError(t, err)
+}
+
+// AssertS3BucketServerSideEncryption checks if the given S3 bucket has a server side encryption configured using the given algorithm and returns an error if it does not
```
Comment (denis256) [inline modules/aws/s3.go:487]: AssertS3BucketServerSideEncryptionE

File: modules/aws/s3.go [hunk 2]
```diff
@@ -478,6 +478,37 @@ func AssertS3BucketPolicyExistsE(t testing.TestingT, region string, bucketName s
...
+	}
+	input := &s3.GetBucketEncryptionInput{
+		Bucket: aws.String(bucketName),
+	}
+	c, err := s3Client.GetBucketEncryption(context.Background(), input)
+	if err != nil {
+		return err
+	}
+
+	err = fmt.Errorf("SSE is not enabled for bucket %s in region %s", bucketName, region)
```
Comment (denis256) [inline modules/aws/s3.go:501]: I suspect this code can be simplified:
```
	for _, rule := range encryption.ServerSideEncryptionConfiguration.Rules {
		if rule.ApplyServerSideEncryptionByDefault == nil {
			continue
		}
		if rule.ApplyServerSideEncryptionByDefault.SSEAlgorithm == algorithm {
			return nil
		}
	}

	return fmt.Errorf("SSE algorithm %s is not enabled for bucket %s in region %s", algorithm, bucketName, region)
```

File: modules/aws/s3.go [hunk 3]
```diff
@@ -478,6 +478,37 @@ func AssertS3BucketPolicyExistsE(t testing.TestingT, region string, bucketName s
 	return nil
 }
 
+// AssertS3BucketServerSideEncryption checks if the given S3 bucket has a server side encryption configured using the given algorithm and fail the test if it does not
+func AssertS3BucketServerSideEncryption(t testing.TestingT, region string, bucketName string, algorithm types.ServerSideEncryption) {
```
Comment (denis256) [inline modules/aws/s3.go:482]: Unused function 'AssertS3BucketServerSideEncryption' - may be a problem if at some point, the function is removed and there will not be any test that fails

**Comment by james03160927** [comment]
PTAL @denis256. Addressed your comments.

**Pull request closed without merging**